dashboard/test/models/lesson_test.rb

```diff
@@ -935,4 +935,141 @@ def create_swapped_lockable_lesson
+      assert_equal @original_lesson.vocabularies.map(&:word), copied_lesson.vocabularies.map(&:word)
+      assert_equal 2, copied_lesson.objectives.length
+      assert_equal @original_lesson.objectives.map(&:description), copied_lesson.objectives.map(&:description)
+      assert_equal @original_lesson.standards, copied_lesson.standards
+      assert_equal @original_lesson.opportunity_standards, copied_lesson.opportunity_standards
+      assert_equal @original_lesson.programming_expressions, copied_lesson.programming_expressions
+    end
+
+    test "can clone lesson with duplicated resources and vocab into another script" do
+      create :resource, name: 'resource1', course_version: @original_course_version, lessons: [@original_lesson]
+      create :vocabulary, word: 'word one', course_version: @original_course_version, lessons: [@original_lesson]
+
+      @destination_script.expects(:write_script_json).once
+      destination_resource = create :resource, name: 'resource1', course_version: @destination_course_version
+      destination_vocab = create :vocabulary, word: 'word one', course_version: @destination_course_version
+      copied_lesson = Lesson.copy_to_script(@original_lesson, @destination_script)
+      assert_equal @destination_script, copied_lesson.script
+      assert_equal [destination_resource], copied_lesson.resources
+      assert_equal [destination_vocab], copied_lesson.vocabularies
+    end
+
+    test "can clone lesson into another script with lessons" do
+      lesson_activity = create :lesson_activity, lesson: @original_lesson
+      activity_section = create :activity_section, lesson_activity: lesson_activity
+      level1 = create :maze, name: 'level 1'
+      create :script_level, script: @original_script, lesson: @original_lesson, levels: [level1],
+        activity_section: activity_section, activity_section_position: 1
+
+      existing_lesson = create :lesson, script: @destination_script, lesson_group: @destination_lesson_group, has_lesson_plan: true
+      existing_lesson_activity = create :lesson_activity, lesson: existing_lesson
+      existing_activity_section = create :activity_section, lesson_activity: existing_lesson_activity
+      level2 = create :maze, name: 'level 2'
+      create :script_level, script: @destination_script, lesson: existing_lesson, levels: [level2],
+        activity_section: existing_activity_section, activity_section_position: 1
+
+      @destination_script.expects(:write_script_json).once
+      copied_lesson = Lesson.copy_to_script(@original_lesson, @destination_script)
+      @destination_script.reload
+      assert_equal @destination_script, copied_lesson.script
+      assert_equal 2, @destination_script.script_levels.length
+      assert_equal [level2, level1], @destination_script.script_levels.map(&:level)
+      assert_equal 1, copied_lesson.script_levels.length
+      assert_equal [level1], copied_lesson.script_levels.map(&:level)
```

Comment:
I believe this verifies the script level's `lesson_id`. Please also verify the `activity_section_id`, `activity_section_position` and `script_id` of each script level. 

we do have `validate_activity_section_lesson` and `validate_activity_section_position`, which may be getting run, but unfortunately they will allow the case where `activity_section_id` is null (or possibly invalid).

consider this to be interest paid on the tech debt of having redundant ownership and position values on our models 😬     I do think this will be easier to clean up once all scripts are migrated, though. https://codedotorg.atlassian.net/browse/PLAT-192